torch/csrc/lazy/core/lazy_graph_executor.cpp

```diff
@@ -635,7 +641,7 @@ LazyGraphExecutor::SyncTensorCollection LazyGraphExecutor::CollectSyncTensors(
         tensors[i]->CurrentDataHandle() == nullptr) {
       Value ir_value = tensors[i]->CurrentIrValue();
       if (ir_value) {
-        if (getBackend()->ShouldSyncTensor(tensors[i])) {
+        if (ShouldSyncTensor(tensors[i])) {
```

Comment:
Is there a reason why it takes a `tensor[i]` isntead of the `ir_value`?

Reply:
This is because we sometimes filter whether or not a tensor should be synced based on the tensor id. So, we need the tensor. The default implementation is simply just extracting the IR value from the tensor and making sure its not an unsupported op, same as before.

If there is a better way to filter the tensor id based solely on the IR value, I'd be happy to change it back. But as far as I can tell, there is no way to get the tensor id back from just the IR value